### PR TITLE
Fix spacing on list collectors

### PR DESCRIPTION
**Pull request by rmccar**
### What is the context of this PR?
Extra space being added to list collectors with and without items.
- For ones without items an empty summary is being added which is removed in this PR
- For ones with items a double margin is being added (to the top of the summary and bottom of the question) which is also removed in this PR

This PR has also fixed a inconsistency in the size of the indentation in the sectionsummary.html file 

### How to review 
- Using the test_list_collector schema compare the differences between master and this branch and see that the gaps have been reduced in both instances

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by rmccar** [comment]
[Fix spacing on list collectors](https://trello.com/c/jgnc6TQP/5057-fix-spacing-on-list-collectors)

**Review by petechd — APPROVED** [review]

File: templates/partials/question.html [hunk 1]
```diff
@@ -93,9 +93,11 @@
 }) %}
   {%- if content.list -%}
     {% set list = content.list %}
-    <div class="ons-u-mb-m">
-       {% include 'partials/summary/list-summary.html' %}
-    </div>
+    {%- if content.list.list_items | length > 0 -%}
```
Comment (LJBabbage) [inline templates/partials/question.html:96]: Not sure why you would have 2 separate conditions, I don't see the benefit of the set if there are no list_items. Empty list items are also considered false so you could just have this
```  
{%- if content.list and content.list.list_items -%}
   {% set list = content.list %}
    <div class="ons-u-mb-m">
        {% include 'partials/summary/list-summary.html' %}
    </div>
{% endif %}
```

**Review by LJBabbage — CHANGES_REQUESTED** [review]

File: templates/partials/question.html (hunk 1 above)
Comment (rmccar) [reply templates/partials/question.html]: Yeah I agree, I've fixed that. I originally tried having one condition in a slightly different way and it fell over but that seems to work

File: templates/partials/question.html [hunk 2]
```diff
@@ -91,10 +91,10 @@
   "instruction": question_instruction,
   "legendIsQuestionTitle": should_wrap_with_fieldset(question)
 }) %}
-  {%- if content.list -%}
+  {%- if content.list and content.list.list_items and content.list.list_items | length > 0 -%}
```
Comment (LJBabbage) [inline templates/partials/question.html:94]: Not sure why you have checked for the existence of content.list.list_items, then its length? content.list.list_items would just return false if empty so you don't need the length check

**Pull request merged**